src/Features/Core/Portable/MetadataAsSource/AbstractMetadataAsSourceService.DocCommentFormatter.cs

```diff
@@ -200,11 +207,33 @@ internal static ImmutableArray<string> Format(IDocumentationCommentFormattingSer
                         firstInLine = true;
                     }
 
-                    current = end;
+                    span = span.Slice(end);
                 }
 
                 formattedLines.Add(sb.ToString());
             }
+
+            public static ImmutableArray<ReadOnlyMemory<char>> Split(
```

Comment:
Consider making an extension. Consider returning an IEnumerable

Reply:
Note: on net-8 something, this method exists right?  So maybe use that, and polyfill it in if on net standard?

Reply:
I intentionally made it return a type with a struct enumerator. Let me know if that's generally not preferred in this scenario.

As for net-8, there are Split extensions, but I don't think I'm fully understanding those. It looks like they require a Span argument in which to store the results, but I'm not sure how large to make that.

Reply:
As for making this an extension method, I thought about that too, but it looks like the other extension methods on ReadOnly(Memory/Span) are down in the compiler, which the features assembly doesn't have IVT to.

Reply:
We don't need ivt. A common thing we do is just link the same files into the ide layer.

Reply:
> Let me know if that's generally not preferred in this scenario.

It's basically a tradeoff. One form allocatesa linear sequence up front, but has free enumeration.  

The other form allocates for enumeration, but doesn't need the contiguous initial allocation. 

I'm personally fine with either.

Reply:
> As for net-8, there are Split extensions, but I don't think I'm fully understanding those. It looks like they require a Span argument in which to store the results, but I'm not sure how large to make that.

Ah. My mistake then. Ignore that idea then :-)

I thought there was some sort of SplitLines helper that streamed results.